src/main/java/mateusz/michal/chat/Service/RegistrationService.java

```diff
@@ -42,71 +62,23 @@ private User createUser(UserDTO userDTO){
         return user;
     }
 
-    private boolean isNameNotPresent(String name){
-        return name.equals("");
-    }
-
-    private boolean isNameIncorrect(String name){
-        return (name.startsWith(" ") || name.endsWith(" ") || hasNameSpecialCharacters(name));
-    }
-
-    private boolean isEmailNotPresent(String email){
-        return email.equals("");
-    }
-
-    private boolean isPassworNotPresent(String password){
-        return password.equals("");
-    }
-
-    private boolean hasNameSpecialCharacters(String name){
-        for(char c : name.toCharArray()){
-            if(!(Character.isLetterOrDigit(c))){
-                return true;
-            }
-        }
-        return false;
-    }
-
-    private boolean isNameNull(String name){
-        return name == null;
-    }
-
-    private boolean isEmailNull(String email){
-        return email == null;
-    }
-
-    private boolean isPasswordNull(String password){
-        return password == null;
-    }
-
-    private boolean isUserInDatabaseByName(String name){
-        Optional<User> user = Optional.ofNullable(userRepository.findByName(name));
-        return user.isPresent();
-    }
-
-    private boolean isUserInDatabaseByEmail(String email){
-        Optional<User> user = Optional.ofNullable(userRepository.findByEmail(email));
-        return user.isPresent();
-    }
-
-    private boolean isEmailIncorrect(String email){
-        Pattern pattern = Pattern.compile("[a-zA-Z]+[\\.a-zA-Z0-9]*@[a-zA-Z0-9]+\\.[a-z]{2,}[a-z]*");
-        return !pattern.matcher(email).matches();
-    }
-
-    private boolean isPasswordIncorrect(String password) {
-        Pattern pattern = Pattern.compile("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z])" +
-                "(?=.*[\\!\\@\\#\\$\\%\\^\\&\\*])(?!=.*\\s).{8,15}");
-        return !pattern.matcher(password).matches();
+    // method that validate request form
+    private List<MyError> validateRegistrationRequest(UserDTO userDTO){
+        List<MyError> errors = new ArrayList<>();
+        errors.addAll(validateUserName(userDTO.getName()));
+        errors.addAll(validateUserEmail(userDTO.getEmail()));
+        errors.addAll(validateUserPassword(userDTO.getPassword()));
+        return errors;
     }
 
+    // methods that validate each property
     private List<MyError> validateUserName(String name){
         List<MyError> errors = new ArrayList<>();
         if(isNameNull(name)){
             errors.add(new MyError(400,RegisterFormErrorCode.NAME_NULL,
                     "parameter name is null"));
         } else {
-            if(isNameNotPresent(name)){
+            if(isNameEmptyString(name)){
```

Comment:
nit: tak naprawdę, wydaje mi się że wystarczyłaby jedna metoda `isEmptyString` 🤔 podobnie jak `isNull` 😃 aczkolwiek nie jestem pewien czy to nie jest sprzeczne z założeniami Twojej architektury :)

Reply:
Na początku walidacja parametrów była w jednej metodzie: wywołanie w jednej metodzie dwa razy metody isNull albo isEmptyString nie byłoby zbyt czytelne. 
Teraz kiedy walidacja została rozbita na poszczególne pola: Name, Email, Password można zastąpić każdą z trzech metod zwykłą metodą isNull. Jednak nie wiem czy jest sens aż tak to rozdrabniać